sdks/how-to/how-to-authenticate-a-user.md

```diff
@@ -0,0 +1,488 @@
+```typescript
+const createdPatient = await authenticatedApi.patientApi.createOrModifyPatient(new Patient({
+  firstName: 'John',
+  lastName: 'Snow',
+  gender: 'male',
+  note: 'Winter is coming'
+}))
+```
+
+<details>
+    <summary>Output</summary>
+
+```json
+{
+  "id": "48e571a0-ac5f-47b3-8e25-16f5e78b50c9",
+  "identifiers": [],
+  "labels": {},
+  "codes": {},
+  "names": [
+    {
+      "firstNames": [
+        "John"
+      ],
+      "prefix": [],
+      "suffix": [],
+      "lastName": "Snow",
+      "text": "Snow John",
+      "use": "official"
+    }
+  ],
+  "languages": [],
+  "addresses": [],
+  "mergedIds": {},
+  "active": true,
+  "deactivationReason": "none",
+  "partnerships": [],
+  "patientHealthCareParties": [],
+  "patientProfessions": [],
+  "parameters": {},
+  "properties": {},
+  "rev": "1-8e3ad0d7e3179188dcd95f186f78b68d",
+  "created": 1664552695128,
+  "modified": 1664552695128,
+  "author": "3363719b-579e-4640-ac62-13e608e69395",
+  "responsible": "d6c8dbc7-eaa8-4c95-b9b3-920fb70ce59b",
+  "firstName": "John",
+  "lastName": "Snow",
+  "gender": "male",
+  "birthSex": "unknown",
+  "personalStatus": "unknown",
+  "note": "Winter is coming",
+  "systemMetaData": {
+    "hcPartyKeys": {},
+    "privateKeyShamirPartitions": {},
+    "secretForeignKeys": [],
+    "cryptedForeignKeys": {},
+    "delegations": {
+      "d6c8dbc7-eaa8-4c95-b9b3-920fb70ce59b": {}
+    },
+    "encryptionKeys": {
+      "d6c8dbc7-eaa8-4c95-b9b3-920fb70ce59b": {}
+    },
+    "aesExchangeKeys": {},
+    "transferKeys": {}
+  }
+}
+```
+
+</details>
+
+
+But what do you have to do when Daenaerys will need to login again in a few days ?
+
+# Login a user
+In iCure, the login flow is extremely similar to the registration.
+Once Daenaerys token is expired, she will need to identify herself again by starting a new authentication process. 
+
+As Daenaerys is not authenticated anymore, you have to create a new AnonymousMedTechApi instance. 
+
+<!-- file://code-samples/how-to/authenticate-user/index.mts snippet:Login-->
+```typescript
+const anonymousApiForLogin = await new AnonymousMedTechApiBuilder()
+  .withICureUrlPath(iCureUrl)
+  .withCrypto(webcrypto as any)
+  .withMsgGtwUrl(msgGtwUrl)
+  .withMsgGtwSpecId(specId)
+  .withAuthProcessByEmailId(authProcessByEmailId)
+  .withAuthProcessBySmsId(authProcessBySmsId)
+  .build()
+
+const authProcessLogin = await anonymousApiForLogin.authenticationApi.startAuthentication(
+  masterHcpId,
+  'Daenerys',
+  'Targaryen',
+  recaptcha,
+  false,
+  userEmail // The email address used for user registration
+)
+```
+
+<details>
+    <summary>Output</summary>
+
+```json
+
+```
+
+</details>
+
+Daenaerys then receives a new validation code by email.
+
+As you already created an RSA keypair for her, you just need to provide it into the `completeAuthentication` service, 
+getting it back from where you stored it previously. 
```

Comment:
```suggestion
Since you already created an RSA keypair for her, you just need to retrieve it from where you stored it previously
and provide it to the `completeAuthentication` method. 
```